### PR TITLE
trigger validation only if we have currentId

### PR DESCRIPTION
**Pull request by haroun**
## Summary

Fix error with array in vue production build

For future us, please also check https://github.com/apostrophecms/apostrophe/pull/4569

## What are the specific steps to test this change?

To enable the vue dev tools in production

```javascript
const apps = Array.from(document.querySelectorAll('*')).filter((e) => e.__vue_app__).map(e => e.__vue_app__)
const version = apps.at(0).version
const devtools = window.__VUE_DEVTOOLS_GLOBAL_HOOK__
devtools.enabled = true
apps.forEach(app => devtools.emit('app:init', app, version, {}))
```
1. `NODE_ENV=production npm run build && NODE_ENV=production node app.js`
2. Test with starting from an empty array and try to add an item
3. Test with an existing array and trigger a validation error, fix the validation error and then try to delete the item

## What kind of change does this PR introduce?
*(Check at least one)*

- [x] Bug fix
- [ ] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [ ] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

**Review by boutell — APPROVED** [review]
Thank you

File: modules/@apostrophecms/schema/ui/apos/logic/AposArrayEditor.js [hunk 1]
```diff
@@ -142,7 +142,12 @@ export default {
     this.modal.active = true;
     await this.evaluateExternalConditions();
     if (this.next.length) {
-      await this.select(this.next[0]._id);
+      this.currentId = this.next[0]._id;
+      this.currentDoc = {
+        hasErrors: false,
+        data: this.next.find(item => item._id === this.currentId)
+      };
+      this.evaluateConditions();
```
Comment (ValJed) [inline modules/@apostrophecms/schema/ui/apos/logic/AposArrayEditor.js:150]: could be in a method `setCurrentDoc`.

File: modules/@apostrophecms/schema/ui/apos/logic/AposSchema.js [hunk 2]
```diff
@@ -120,7 +125,7 @@ export default {
               required
             },
             value: {
-              data: this.modelValue[item.name]
+              data: this.modelValue[item.name] || {}
```
Comment (ValJed) [inline modules/@apostrophecms/schema/ui/apos/logic/AposSchema.js:128]: Are we  sure data is always supposed to be an object here?
Should be `this.modelValue.data[item.name]`. Looks like it's not used we should verify and create a ticket if needed.

**Review by ValJed — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/schema/ui/apos/logic/AposArrayEditor.js [hunk 3]
```diff
@@ -155,19 +155,22 @@ export default {
...
   },
   methods: {
+    setCurrentDoc(_id) {
+      this.currentId = _id;
+      this.currentDoc = {
+        hasErrors: false,
+        data: this.next.find(item => item._id === _id)
+      };
+      this.evaluateConditions();
+      this.triggerValidation = false;
```
Comment (ValJed) [inline modules/@apostrophecms/schema/ui/apos/logic/AposArrayEditor.js:165]: I would have kept this line out but detail.

**Review by ValJed — APPROVED** [review]

**Pull request merged**